recipes/default.rb

```diff
@@ -22,10 +22,14 @@
 package auditd_package_name_for(node['platform_family'])
 
 service 'auditd' do
-  if platform_family?('rhel') && node['init_package'] == 'systemd'
+  if platform_family?('rhel') && node['init_package'] == 'systemd' && node['platform_version'] < '7.5'
```

Comment:
The logic here could get simplified to reduce the number of checks we have to do.

```ruby
if platform_family?('rhel') && node['init_package'] == 'systemd'
  if node['platform_version'] < '7.5'
    reload_command '/usr/libexec/initscripts/legacy-actions/auditd/reload'
    restart_command '/usr/libexec/initscripts/legacy-actions/auditd/restart'
  else
    reload_command '/usr/sbin/service auditd reload'
    restart_command '/usr/sbin/service auditd restart'
  end
end
```


Also I believe that /usr/sbin/service auditd reload is the default so it's actually quite likely you can just just leave out the reload/restart commands on newer RHEL and let Chef just do the right thing. You should give that a try.
```ruby
if platform_family?('rhel') && node['init_package'] == 'systemd' && node['platform_version'] < '7.5'
  reload_command '/usr/libexec/initscripts/legacy-actions/auditd/reload'
  restart_command '/usr/libexec/initscripts/legacy-actions/auditd/restart'
end
```

Reply:
@tas50 as per the bug listed in the [link here](https://bugzilla.redhat.com/show_bug.cgi?id=1647521), the `auditd` service is the exception to this rule and needs special attention to start/stop/reload properly.

Reply:
Also, the reason I have the `if else` is because RHEL versions prior to 7.5 use the service command to restart, whereas 7.5 and greater use the script method, I wanted to make sure to catch both RHEL types.

Reply:
Can you stick a comment near that 7.5 check that explains why we're doing that. Otherwise someone is going to see it in a year and nuke it because it seems really out of place.

Reply:
done!